components/builder-jobsrv/src/scheduler_datastore/test.rs

```diff
@@ -0,0 +1,358 @@
+#[cfg(test)]
+#[cfg(feature = "postgres_tests")]
+// cargo test --features postgres_tests to enable
+// from root
+// cargo test -p habitat_builder_jobsrv --features=postgres_tests
+// --manifest-path=components/builder-jobsrv/Cargo.toml
+mod test {
```

Comment:
I think this will be `scheduler_datastore::test::test` because of how it's nested. You probably want to put the `test` and `feature` annotations on the declaration up in `scheduler_datastore`.

The test-only structs from `scheduler_datastore` could also just as well be defined in here.

Alternatively, all this could go into a `tests` module in `scheduler_datastore`... up to you :smile:

Reply:
This is structured this way because we 1) wanted to to be in a separate file
and 2) wanted to have the cfg feature applied to it, which only seems to work right
directly in front of a mod declaration. Couldn't quite figure out how to get that to work without the double layering you see here.